discretisedfield/mesh.py

```diff
@@ -282,57 +282,60 @@ def subregions(self):
 
         Returns
         -------
-        dict
+        dict, None
 
             A dictionary defining subregions in the mesh. The keys of the
             dictionary are the region names (``str``) as valid Python variable
             names, whereas the values are ``discretisedfield.Region`` objects.
-            Defaults to an empty dictionary.
+            Defaults to None.
         """
         return self._subregions
 
     @subregions.setter
     def subregions(self, subregions):
-        if not isinstance(subregions, dict):
-            raise TypeError(
-                "Subregions must be a dictionary relating the name of a subregion with"
-                " its region."
-            )
-
-        if not all(isinstance(key, str) for key in subregions):
-            raise TypeError("The keys of subregion dictionary must be strings.")
-
-        # Check if subregions are aligned with the mesh
-        for key, value in subregions.items():
-            # Is the subregion in the mesh region?
-            if value not in self.region:
-                msg = f"Subregion {key} is not in the mesh region."
-                raise ValueError(msg)
+        if subregions is None:
+            self._subregions = None
+        else:
```

Comment:
I maybe was not precise enough here. I would keep the empty dict for the subregions if there are no subergions. However, the default value in `__init__` should not be mutable. So something like:
```python
if subregions is None:
    subregions = {}

Reply:
We can then presumably avoid the `else` because an empty dict should not cause any problems.